core/src/main/java/com/github/mizool/core/rest/errorhandling/GenericErrorMapper.java

```diff
@@ -84,11 +84,19 @@ private Throwable determineRootCause(Throwable t)
         return rootCause;
     }
 
-    private Map<String, String> createExceptionParameters(Throwable throwable)
+    private Map<String, Object> createExceptionParameters(Throwable throwable)
     {
-        Map<String, String> parameters = Maps.newHashMap();
+        Map<String, Object> parameters = Maps.newHashMap();
         parameters.put("Exception", throwable.getMessage());
         parameters.put("RootCause", Throwables.getRootCause(throwable).getMessage());
+        if (throwable instanceof ParameterizedThrowable)
+        {
+            Map<String, Object> exceptionParameters = ((ParameterizedThrowable) throwable).getExceptionParameters();
+            if (exceptionParameters != null)
```

Comment:
I was thinking about whether we should allow the parameters to be `null`. Since the developer made the effort to implement `ParameterizedThrowable`, it could be safe to assume that `null` parameters are some kind of `CodeInconsistencyException`.

However, I really don't want to add exception cases to the exception handling, so perhaps this is the solution with the least amount of pain down the road.

What do you think @bannmann ?

Reply:
I agree. Further rationale: the main objective is to transport the original error. So even if we suspect that it is likely lacking some details, we should not *replace* that error with a `CodeInconsistencyException`.

Also, I wouldn't bother *adding* information about that case because the client / end user is likely not able to fix it, and it's not their business anyway.

As an alternative, I briefly considered adding a log warning here, but decided against it to avoid log spam in production systems. If the lack of details is relevant, (frontend) developers will surely notice the error anyway.